pkg/adapterManager/manager.go

```diff
@@ -96,11 +96,17 @@ func (m *Manager) Execute(cfg *aspect.CombinedConfig, attrs attribute.Bag, mappe
 	}
 
 	var asp aspect.Wrapper
-	var err error
 	if asp, err = m.cacheGet(cfg, mgr, adapter); err != nil {
 		return nil, err
 	}
 
+	defer func() {
```

Comment:
We should wrap *all* AdapterImpl code in a try-catch.

asp.Execute and  m.CacheGet both call AdapterImpl supplied code. 
pl. move defer before the cacheGet call.